rcl/test/rcl/test_node.cpp

```diff
@@ -511,15 +511,21 @@ TEST_F(CLASSNAME(TestNodeFixture, RMW_IMPLEMENTATION), test_rcl_node_init_with_i
     EXPECT_EQ(RCL_RET_ERROR, ret);
     rcl_reset_error();
   }
+
   // Try normal init but force an internal error on fini.
   {
     ret = rcl_node_init(&node, name, namespace_, &context, &options);
     EXPECT_EQ(RCL_RET_OK, ret);
-    auto mock = mocking_utils::inject_on_return("lib:rcl", rmw_destroy_node, RMW_RET_ERROR);
-    ret = rcl_node_fini(&node);
-    EXPECT_EQ(RCL_RET_ERROR, ret);
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rcl", rcl_logging_rosout_fini_publisher_for_node, RCL_RET_ERROR);
```

Comment:
Unfortunately, this does not work on macOS; according to @hidmic, the problem is that intra-library calls use relative calls, and thus we cannot patch it this way.

Alternately, I also tried using the RCUTILS_FAULT_INJECTION framework.  But in brief try here, I also couldn't get that to work properly.

At this point, I may end up just disabling this test on CentOS.  I actually don't know quite how to do *that* either, but at least we keep the test as-is for the Tier 1 platforms and just disable it there.

Reply:
Hmm, perhaps it's not the cleanest and prettiest solution, but what if we add a macro or a function to check if `inject_on_return()` is supported so we can `GTEST_SKIP()` the entire test if not?

Reply:
> Hmm, perhaps it's not the cleanest and prettiest solution, but what if we add a macro or a function to check if `inject_on_return()` is supported so we can `GTEST_SKIP()` the entire test if not?

I did something slightly more coarse in f7577f5 .  There, I go and detect the Linux distribution, and if it is "centos", I just skip the test.  On everything that is not Linux, or does not have `/etc/os-release`, I just assume that we should compile and run it.  What do you think about that approach?

Reply:
It looks good. What I had in mind was a bit more general, but I don't really know if I can extrapolate what we know about Mimick to any platform.